### PR TITLE
[RFC6265bis] Fix ABNF.

### PR DESCRIPTION
**Pull request by mikewest**
This patch incorporates two ABNF changes:

1.  Closes https://github.com/httpwg/http-extensions/issues/1074 by
    adjusting the `cookie-pair` grammar to include optional spaces.
    We test this behavior in a few distinct WPTs, including
    chromium0014, name0018, value0006.

2.  Closes https://github.com/httpwg/http-extensions/issues/1119
    by removing OWS from the spec, as user agents generally follow
    the "SHOULD" in the current text to limit themselves to a single
    SP rather than serializing with OWS. It's reasonable to lock that
    in, and it also happens to match all the `Cookie:` header tests
    we have.

Thanks to both @reschke and @bakulf for paying attention to this.

### CODE REVIEW

**Comment by mikewest** [comment]
@reschke and @bakulf: Would you mind taking a look at this patch?

**Comment by mikewest** [comment]
Adding @johnwilander as well.

**Comment by mikewest** [comment]
I'm planning on merging this tomorrow unless someone objects or has suggestions. :)

File: draft-ietf-httpbis-rfc6265bis.md [hunk 1]
```diff
@@ -466,14 +455,13 @@ grammar:
 ~~~ abnf
 set-cookie-header = "Set-Cookie:" SP set-cookie-string
 set-cookie-string = cookie-pair *( ";" SP cookie-av )
-cookie-pair       = cookie-name "=" cookie-value
-cookie-name       = token
+cookie-pair       = cookie-name *[WSP] "=" *[WSP] cookie-value
+cookie-name       = 1*cookie-octet
```
Comment (bakulf) [inline draft-ietf-httpbis-rfc6265bis.md:459]: This is not enough.
What about these:

1. `Set-Cookie: a=b ;   sameSite=strict`
2. `SetCookie: a=b ; sameSite = strict`
3. `SameSite:  a=b; sameSite=lax ; path = /`

Comment (reschke) [reply draft-ietf-httpbis-rfc6265bis.md:459]: *[WSP]

is the same as

    *WSP

Comment (mikewest) [reply draft-ietf-httpbis-rfc6265bis.md:461]: Thanks to you both. I've replaced WSP with OWS (I'd mistakenly assumed RFC7230's OWS contained a wider character set than WSP), and added more flexibility around delimiters, in line with the WPT suite's tests.

WDYT?

**Comment by mikewest** [comment]
> if you made sure that the new places for OWS are actually working in practice

I believe these are covered by the parsing tests in WPT (https://github.com/web-platform-tests/wpt/tree/master/cookies/http-state/resources/test-files)

**Comment by reschke** [comment]
...alternative would be "BWS", if you want to allow but discourage.

**Comment by mikewest** [comment]
Ah. That might be a reasonable idea. Does `BWS` have any normative text associated with it?

**Comment by reschke** [comment]
> The BWS rule is used where the grammar allows optional whitespace only for historical reasons. A sender MUST NOT generate BWS in messages. A recipient MUST parse for such bad whitespace and remove it before interpreting the protocol element.

https://greenbytes.de/tech/webdav/rfc7230.html#rfc.section.3.2.3.p.4

**Comment by mikewest** [comment]
Merging this. Happy to take suggestions for further cleanup as we go.

**Review by reschke — COMMENTED** [review]
FWIW, when I approved this, I was focusing on the WS definition from the point of view of the HTTP spec.

Obviously changing the allowed set of character names is problematic.

File: draft-ietf-httpbis-rfc6265bis.md (hunk 1 above)
Comment (reinwowa47-ship-it) [reply draft-ietf-httpbis-rfc6265bis.md:462]: Y

**Pull request merged**